reconcile/typed_queries/tekton_pipeline_providers.py

```diff
@@ -0,0 +1,21 @@
+from collections.abc import Callable
+from typing import Optional
+
+from reconcile.gql_definitions.common.pipeline_providers import (
+    PipelinesProviderTektonV1,
+    query,
+)
+from reconcile.utils import gql
+
+
+def get_tekton_pipeline_providers(
+    query_func: Optional[Callable] = None,
+) -> list[PipelinesProviderTektonV1]:
+    if not query_func:
+        query_func = gql.get_api().query
+    providers: list[PipelinesProviderTektonV1] = []
+    for provider in query(query_func).pipelines_providers or []:
+        if not isinstance(provider, PipelinesProviderTektonV1):
+            continue
+        providers.append(provider)
+    return providers
```

Comment:
This is equivalent to
```python
pipelines_providers = query(query_func).pipelines_providers or []
return [p for p in pipeline_providers if isinstance(p, PipelinesProviderTektonV1)]
```

Reply:
Good catch!
The original intent of this is to convert the existing logic towards typed queries. I.e., Im more inclined to keep the original structure for now.